src/formatters/SelectCellFormatter.tsx

```diff
@@ -1,33 +1,31 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
 import clsx from 'clsx';
 
-type SharedInputProps = Pick<React.InputHTMLAttributes<HTMLInputElement>,
-  | 'disabled'
-  | 'aria-label'
-  | 'aria-labelledby'
->;
-
-export interface SelectCellFormatterProps extends SharedInputProps {
+export interface SelectCellFormatterProps {
   value: boolean;
+  tabIndex?: number;
+  isCellSelected?: boolean;
+  disabled?: boolean;
   onChange: (value: boolean, isShiftClick: boolean) => void;
 }
 
-export function SelectCellFormatter({
-  value,
-  disabled,
-  onChange,
-  'aria-label': ariaLabel,
-  'aria-labelledby': ariaLabelledBy
-}: SelectCellFormatterProps) {
+export function SelectCellFormatter({ value, tabIndex, isCellSelected, disabled, onChange }: SelectCellFormatterProps) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  useEffect(() => {
+    if (isCellSelected) {
```

Comment:
User can handle focus however they like and for complex components `event` can be stopped from propagating.